pkg/alertmanager/operator.go

```diff
@@ -927,6 +927,24 @@ func checkAlertmanagerConfig(ctx context.Context, amc *monitoringv1alpha1.Alertm
 			}
 		}
 
+		for j, sConfig := range receiver.SlackConfigs {
```

Comment:
A request: 
Can we have this function checkAlertmanagerConfig be refactored a bit to call two funcs checkAlertManagerRecievers and checkAlertManagerRoutes?

In checkAlertManagerReceivers, each inner for loop can be a separate smaller func.

Feel free to ignore this comment if its not suitable.

Reply:
I think splitting the check for the receivers makes sense but in my opinion, this should be done in another PR dedicated to that change. I can work on it next week

Reply:
@nrchakradhar I split the loop in smaller functions as requested :)